src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpTagHelperRunnerInitializationVisitor.cs

```diff
@@ -26,6 +26,25 @@ public class CSharpTagHelperRunnerInitializationVisitor : CodeVisitor<CSharpCode
             _tagHelperContext = Context.Host.GeneratedClassContext.GeneratedTagHelperContext;
         }
 
+        /// <inheritdoc />
+        public override void Accept(Chunk chunk)
```

Comment:
Good point :+1: . One thing to note though: With the current approach you'll be diving into every attribute (`DynamicCodeAttributeChunk`, `LiteralCodeAttributeChunk`, `CodeAttributeChunk`) and expression blocks (`ExpressionBlockChunk`) which can't/shouldn't contain directives or `TagHelper`s.

Not horrible but not ideal. It might be good to make a base class that has knowledge about diving into known Razor blocks (template chunks, sections, statements etc.) that can contain directives or complex structures like `TagHelperChunk`s. That way you can just inherit from it in this class and the `CSharpUsingVisitor`.

Since you're updating the `@using` here as well, this problem impacts other directives as well, for instance: any `@model` or `@addTagHelper` inside of sections, statements etc. won't be picked up.